### ISSUE TITLE
false negative missing member

### ISSUE THREAD

**Issue by dpinol** [PyCQA/pylint#5626]
### Bug description

```python
class Class:
    pass


def f(c: Class) -> None:
    c.shouldFail()

c = Class()
c.shouldFail()
```

This code should cause  two E1101 no-member reports, but it only reports the second one.

### Command used

```shell
pylint -E kk.py
```

### Pylint output

```shell
kk.py:9:0: E1101: Instance of 'Class' has no 'shouldFail' member (no-member)
```

### Expected behavior

```
kk.py:6:0: E1101: Instance of 'Class' has no 'shouldFail' member (no-member)
kk.py:9:0: E1101: Instance of 'Class' has no 'shouldFail' member (no-member)

```

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.1
Python 3.9.9 (main, Dec 28 2021, 17:28:48) 
[GCC 11.2.0]
```

### OS / Environment

Can reproduce in both windows and WLS2

### Additional dependencies

On a virtualenv with only pylint installed
```
astroid==2.9.1
isort==5.10.1
lazy-object-proxy==1.7.1
mccabe==0.6.1
platformdirs==2.4.1
pylint==2.12.2
toml==0.10.2
typing_extensions==4.0.1
wrapt==1.13.3
```

**Comment by DanielNoord** [comment]
We would need to understand typing annotations for this to work. See https://github.com/PyCQA/pylint/issues/4813

**Comment by Pierre-Sassoulas** [comment]
Closing in favor of #4813, this depends on a global decision to take type into account or not.
